### PR TITLE
Update all gtk crates to 18.x

### PR DESCRIPTION
**Pull request by atlanticaccent**
Updates all GTK dependencies from 0.16.x to 0.18.x.

Fixes breaking changes introduced.
Also adds a top level allow for clippy::arc_with_non_send_sync as it's triggered pretty much everywhere in the project and is a massive undertaking to fix.

Currently using a git dependency for piet as it does not have a release including updates to its GTK dependencies.

### CODE REVIEW

**Comment by atlanticaccent** [comment]
I'm happy to update this PR yes!

I'll need to check if updating to GTK 0.19 will break anything I'm working on (I have a bunch of other dependencies with GTK crates in their deps as well) so will default to only going to 0.18 for now.

File: druid-shell/Cargo.toml [hunk 1]
```diff
@@ -100,14 +100,14 @@ bitflags = "1.3.2"
 [target.'cfg(any(target_os = "freebsd", target_os="linux", target_os="openbsd"))'.dependencies]
 ashpd = { version = "0.3.2", optional = true }
 # TODO(x11/dependencies): only use feature "xcb" if using X11
-cairo-rs = { version = "0.16.7", default-features = false, features = ["xcb"] }
-cairo-sys-rs = { version = "0.16.3", default-features = false, optional = true }
+cairo-rs = { version = "0.18.0", default-features = false, features = ["xcb"] }
+cairo-sys-rs = { version = "0.18.0", default-features = false, optional = true }
```
Comment (bilelmoussaoui) [inline druid-shell/Cargo.toml:104]: you don't need to explicitly depend on this, you can get it from `cairo::ffi`

File: druid-shell/Cargo.toml [hunk 2]
```diff
@@ -100,14 +100,14 @@ bitflags = "1.3.2"
...
 futures = { version = "0.3.26", optional = true, features = ["executor"]}
-gdk-sys = { version = "0.16.0", optional = true }
+gdk-sys = { version = "0.18.0", optional = true }
 # `gtk` gets renamed to `gtk-rs` so that we can use `gtk` as the feature name.
-gtk-rs = { version = "0.16.2", package = "gtk", optional = true }
-glib-sys = { version = "0.16.3", optional = true }
-gtk-sys = { version = "0.16.0", optional = true }
+gtk-rs = { version = "0.18.0", package = "gtk", optional = true }
```
Comment (bilelmoussaoui) [inline druid-shell/Cargo.toml:108]: from gtk, you can go to `gtk::gdk::ffi` for example, all those dependencies are not needed. You could just depend on `gtk` except for cairo where you enable `xcb` feature

File: druid-shell/src/backend/gtk/window.rs [hunk 3]
```diff
@@ -24,12 +24,12 @@ use std::slice;
...
 use std::time::Instant;
 
+use cairo::glib::{Propagation, ControlFlow};
 use gtk::gdk_pixbuf::Colorspace::Rgb;
 use gtk::gdk_pixbuf::Pixbuf;
-use gtk::glib::source::Continue;
 use gtk::glib::translate::FromGlib;
 use gtk::prelude::*;
-use gtk::traits::SettingsExt;
+use gtk::traits::GtkSettingsExt;
```
Comment (bilelmoussaoui) [inline druid-shell/src/backend/gtk/window.rs:32]:
```suggestion
```
already part of the prelude

File: druid-shell/src/backend/gtk/window.rs [hunk 4]
```diff
@@ -427,7 +427,7 @@ impl WindowBuilder {
             .connect_enter_notify_event(|widget, _| {
                 widget.grab_focus();
 
-                Inhibit(true)
+                cairo::glib::Propagation::Stop
```
Comment (bilelmoussaoui) [inline druid-shell/src/backend/gtk/window.rs:430]: you imported it above

**Comment by xStrom** [comment]
Can't move on to 0.20 because `gtk3-rs` has been abandoned and 0.18 was the last release they did before [archiving the repository](https://github.com/gtk-rs/gtk3-rs).

Also, this PR here has been superseded by #2410. If @atlanticaccent or anyone else has interest in the upgrade to 0.18 then please take a look at #2410 and see if it works for you, so we can get that merged.

**Pull request closed without merging**